File: src/java/LinearSearch.java
// DSA01 (Java): Linear Search

import java.util.Scanner;

public class LinearSearch {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the number of elements: ");
        int n = sc.nextInt();

        int[] arr = new int[n];
        System.out.print("Enter the elements: ");
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        System.out.print("Enter the search element: ");
        int target = sc.nextInt();

        System.out.print("The array elements are: ");
        for (int i : arr)
            System.out.print(i + " ");
        System.out.println();

        int pos = linearSearch(arr, target);

        if (pos != -1)
            System.out.println("Element found at index " + pos);
        else
            System.out.println("Element not found");

        sc.close();
    }

    static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target)
                return i;
        }
        return -1;
    }
}

File: src/kotlin/LinearSearch.kt
// DSA01 (Kotlin): Linear Search

import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    print("Enter the number of elements: ")
    val n = sc.nextInt()

    var arr = IntArray(n)
    print("Enter the elements: ")
    for (i in arr.indices)
        arr[i] = sc.nextInt()

    print("Enter the search element: ")
    val target = sc.nextInt()

    print("The array elements are: ")
    for (i in arr)
        print("$i ")
    println()

    val pos = linearSearch(arr, target)

    if (pos != -1)
        println("Element found at index $pos")
    else
        println("Element not found")

    sc.close()
}

fun linearSearch(arr: IntArray, target: Int): Int {
    for (i in arr.indices) {
        if (arr[i] == target)
            return i
    }
    return -1
}
